### ISSUE TITLE
Refactor event type to string literals

### ISSUE THREAD

**Issue by itaied246** [stripe/stripe-node#1456]
### Describe the bug

```typescript
event.type === "payment_intent.created" // correct
event.type === "payment_intent.not_exist_type" // incorrect - should not compile
```

The `Event` interface `type` field should be string literals instead of `string`:

```typescript
/**
 * Description of the event (e.g., `invoice.created` or `charge.refunded`).
 */
type: string;
```

### To Reproduce

```typescript
event.type === "payment_intent.not_exist_type" 
```

### Expected behavior

Running typescript compilation should fail

### Code snippets

```JavaScript
You should create the string literals of the events:


type: "payment_intent.created" | "payment_intent.succeeded" ...
```

### OS

Ubuntu

### Node version

Node v16.15.0

### Library version

stripe-node@9.5.0

### API version

2020-08-27

**Comment by dcr-stripe** [comment]
Hey @itaied246 - thanks for the report. I'll label this as a feature request - I think this is a really interesting idea and something we'll want to consider in our future roadmap.

**Comment by huw** [comment]
This has been fixed in v13.4.0; `event.type` is now a union of string literals!
